app/geant-exporter/GeantPhysicsTableWriter.hh

```diff
@@ -0,0 +1,72 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantPhysicsTableWriter.hh
+//! \brief Write Geant physics tables into a ROOT file
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <memory>
+
+#include "io/GeantPhysicsTable.hh"
+
+class TFile;
+class TTree;
+
+class G4VProcess;
+class G4VEmProcess;
+class G4VEnergyLossProcess;
+class G4VMultipleScattering;
+class G4ParticleDefinition;
+class G4PhysicsTable;
+
+using celeritas::GeantPhysicsTable;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Use an existing TFile address as input to create a new "tables" TTree used
+ * store Geant physics tables
+ *
+ * TFile passed to the constructor must be open with the "recreate" flag so the
+ * class has writing privileges.
+ */
+class GeantPhysicsTableWriter
+{
+  public:
+    // Constructor adds a new "tables" TTree to the existing ROOT TFile
+    GeantPhysicsTableWriter(TFile* root_file);
+    // Default destructor
+    ~GeantPhysicsTableWriter() = default;
+
+    // Write the physics tables from a given particle and physics process
+    // Expected to be called within a G4ParticleTable iterator loop
+    void add_physics_tables(const G4VProcess&           process,
+                            const G4ParticleDefinition& particle);
+
+  private:
+    // Loop over EM processes and write tables to the ROOT file
+    void fill_em_tables(const G4VEmProcess& em_process);
+    // Loop over energy loss processes and write tables to the ROOT file
+    void fill_energy_loss_tables(const G4VEnergyLossProcess& eloss_process);
+    // Loop over multiple scattering processes and write tables to the ROOT
+    // file
+    void
+    fill_multiple_scattering_tables(const G4VMultipleScattering& msc_process);
+    // Write the physics vectors from a given G4PhysicsTable to this->table_
+    void fill_physics_vectors(G4PhysicsTable& table, std::string xs_or_eloss);
+    // Write the remaining elements of this->table_ and fill the tables TTree
+    void fill_tables_tree(G4PhysicsTable& table,
+                          std::string     table_type_name,
+                          std::string     table_name,
+                          std::string     xs_or_eloss);
+
+  private:
+    // TTree created by the constructor
+    std::unique_ptr<TTree> tree_tables_;
+    // Object written in the TTree. Each GeantPhysicsTable is a new TTree entry
+    GeantPhysicsTable table_;
+};
```

Comment:
I've seen these "do not enter" signs on some of your commits, indicating your file doesn't end with a newline, which is odd. Are you editing these in vscode?  Also, again this class should be in a namespace.

Reply:
Yes, vscode with clang-formatting. Sometimes I just put a silly temporary comment at the end of the file as a quick reminder on what to do on that file, maybe before a lunch break, and when I delete the note I mistakenly remove the trailing newline... All are fixed now.